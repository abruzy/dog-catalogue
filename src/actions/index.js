import axios from 'axios';

const FETCHDOGS = () => dispatch => {
  const allCategories = ['Beef', 'Chicken', 'Dessert'];
  const allCategoriesResult = [];

  // eslint-disable-next-line array-callback-return
  allCategories.map(category => {
    axios.get(`https://www.themealdb.com/api/json/v1/1/filter.php?c=${category}`)
      .then(res => {
        allCategoriesResult.push(...res.data.meals);
      }).then(() => {
        dispatch({
          type: 'FETCH_DATA',
          payload: allCategoriesResult,
        });
      });
  });
};

const FETCHDOG = dogId => async dispatch => {
  const data = await axios.get(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${dogId}`);
  dispatch({
    type: 'FETCH_DOG',
    payload: data.data.meals[0],
  });
};

const FILTERDOGLIST = category => async dispatch => {
  const data = await axios.get(`https://www.themealdb.com/api/json/v1/1/filter.php?c=${category}`);
  dispatch({
    type: 'FETCH_BY_CATEGORY',
    payload: data,
  });
};

const CLEAR_DATA = () => dispatch => {
  dispatch({
    type: 'RESET_DATA',
  });
};

export {
  FETCHDOGS, FILTERDOGLIST, FETCHDOG, CLEAR_DATA,
};
