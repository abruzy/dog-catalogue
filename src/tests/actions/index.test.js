import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { FETCH_RECIPE, CLEAR_DATA } from '../../actions/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it('dispatches the correct action and payload', () => {
    store.dispatch(FETCH_RECIPE());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches the correct action and payload', () => {
    store.dispatch(CLEAR_DATA());
    expect(store.getActions()).toMatchSnapshot();
  });
});
