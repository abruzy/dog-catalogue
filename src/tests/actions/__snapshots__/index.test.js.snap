// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions dispatches the correct action and payload 1`] = `Array []`;

exports[`async actions dispatches the correct action and payload 2`] = `
Array [
  Object {
    "type": "RESET_DATA",
  },
]
`;

exports[`async actions dispatches the correct action and payload 3`] = `
Array [
  Object {
    "type": "RESET_DATA",
  },
]
`;
